runtime.go

```diff
@@ -18,7 +18,7 @@ import (
 // ContextualMain calls a main entry point function with a cancellable
 // context via SIGTERM. This should be called once per process so as
 // to not clobber the signals from Notify.
-func ContextualMain[L ILogger](main func(ctx context.Context, args []string, logger L) error, logger L) {
+func ContextualMain[L ZapCompatibleLogger](main func(ctx context.Context, args []string, logger L) error, logger L) {
```

Comment:
I _think_ this is a somewhat breaking change, so if we do this, we may want to announce it as such. You're modifying the generic constraint here to be more intense (require more methods than `ILogger`). Authors of golang modules, in particular, often use `ContextualMain` to wrap their main function and get automatic signal handling. If they were passing in their own bare-bones logger type, their code may no longer compile.

Reply:
Good catch @benjirewis. I had no intention of reading all of the substitutions.

I'd agree with going back to `ILogger` for these contextual main cases (but keeping the changes for "slow goroutine" methods in this file at ZapCompatibleLogger). We don't need the contextual main loggers to be hooked into the registry system. All this stuff actually gets kicked off before we've created a registry.

Reply:
Good call out, seems like it does not need to be a `ZapCompatibleLogger` and wasn't not sure if this was even necessary. Victim of the search and replace lol. Will revert!